Reuse last init docs

mmrj opened the pull request.

narenranjit commented on documentation/run-strategies-template.ejs:
```diff
@@ -25,8 +25,10 @@ You can also [create your own](#create-your-own).
-You can create your own strategy by passing in a function in the `strategy` parameter to the `F.manager.RunManager()` instantiation call. 
+You can create your own strategy by:
+
+* Passing in a function in the `strategy` parameter to the `F.manager.RunManager()` instantiation call. Strategy functions must return a boolean value for whether or not to create a new run. 
```
Not really -- strategy functions should return objects of the form

```
 {
  getRun: function(){},
  reset: function(){}
}

```

The `true/ false` only applies is the new class you create sub-classes the 'conditional creation' strategy like so

```js
var ConditionalStrategy = F.manager.RunManager.strategies.get('conditional-creation')
var myNewStrategy = new ConditionalStrategy(function() {  //return true or false here});
```

narenranjit commented on documentation/run-strategies-template.ejs:
```diff
@@ -25,8 +25,10 @@ You can also [create your own](#create-your-own).
+You can create your own strategy by:
+
+* Passing in a function in the `strategy` parameter to the `F.manager.RunManager()` instantiation call. Strategy functions must return a boolean value for whether or not to create a new run. 
+* Optionally, passing in a `strategyOptions` object with any configuration options that your strategy function may use. 
```
"pass in", not "passing in" right? I'd reword as:

Some strategies have options you can specify through `strategyOptions` in the RunManager. If you create your own strategy, this will be passed in to your strategy function constructor.

mmrj replied:
expanded this section

narenranjit commented on src/managers/run-strategies/reuse-last-initialized.js:
```diff
@@ -1,17 +1,38 @@
+/**
+ * The `reuse-last-initialized` strategy looks for the most recent run that matches particular critiera; if it cannot find one, it creates a new run and immediately executes a set of "initialization" operations. 
+ *
+ * This strategy is useful if you have a custom initialization function in your model, and want to make sure it's always executed for new runs. This strategy is also useful if you have a time-based model and always want the run you're operating on to start at a particular step. For example:
```
I'd switch the order of these 2 cases; i.e. "if you have a time-base model" and then "also useful for custom init etc"

mmrj replied:
ordered that way so we could have the two use cases and then the code sample, without a paragraph afterward. but this is fine too. updated.

mmrj commented:
next steps:
 - @mmrj review #192 and make docs updates for that
 - figure out how best to get these changes into that branch.

mmrj commented:
hand-merged relevant pieces (incl. addressing comments) to "sm-docs" branch: https://github.com/forio/epicenter-js-libs/commit/f0a67cc6e88f37aafda334f6a9978bff1c7383ac

closing this without merging.
